### ISSUE TITLE
Fix S2325 FP: making public members static is breaking change

### ISSUE THREAD

**Issue by sm-g** [SonarSource/sonar-dotnet#9756]
### Description

see [CA1822](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1822) also

### Reproducer

```
public class FooService
{
        // S2325 
        public int GetValue(int x) => x + 1;
}


public class Bar
{
        FooService _fooService;

        public void Call()
        {
            // I want to keep such usage
            var value = _fooService.GetValue(5);
        }
}


```
 

### Product and Version

SonarAnalyzer.CSharp 10.15.0.120848

**Comment by mary-georgiou** [comment]
Hello @sm-g and thanks for reporting this.

Unfortunately this is not achievable in a reasonable way based on how the analyzer works.
How can we take this decision really?   From this example to me it seems this is a decision of the individual caller so from my POV only way is to disable the rule :/ 

Possible work-arounds are:
1 - define an interface with the methods you want to be instance methods without them actually accessing the instance  (we don't raise if the method is defined as non-static due to an interface contract)
2 - you could disable the rule or accept the issue.


let me know what you think!

**Comment by DaleStan** [comment]
It seems to me that S2325 should be configurable in the same way as [CA1822](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1822), so the user can decide whether or not to run it on public members.

Alternatively, S2325 could be removed entirely since it's redundant with CA1822. Users can enable/configure CA1822 as desired, and you can spend more time on diagnostics Microsoft doesn't already maintain.

**Comment by mary-georgiou** [comment]
Hello @DaleStan, 

Indeed we already follow this practice, and when we implement new rules we aim to not duplicate what is already there on MS side - however once the rule is there, removing it is not that straightforward. Some reasons are that not of all of our users import external issues or not all of them use a Roslyn version where this rule is actually enabled/available.

If CA1822 is available and works well for a specific codebase, I would suggest deactivating S2325 entirely.
